README.md

```diff
@@ -0,0 +1,43 @@
+This repository contains shared modules which will be used for the [translation imbalances](https://meta.wikimedia.org/wiki/Research:Content_Translation_language_imbalances)
+research project.
+
+Module structure
+===
+Each data source is managed by a dedicated python module.  For example, the
```

Comment:
What happens when we have a function that could work with several data sources that are associated by the type of data queried? For example, the Wikipedia APIs and the Wikipedia Dumps are two different data sources, but if we are fetching the pageviews counts (which we can access via both sources), the data structure we aim to achieve may be similar, or it would just make sense to have a module that addresses both versions. I think that we should use a more generic or abstract concept to define this separation.

Reply:
Also, if we can reuse certain functions with data sources that may not be obviously connected, this should be useful too, as we don't want to be redundant by rewriting the same logic or code with a very similar semantic value/meaning.

Reply:
Excellent critiques, thanks!  The point I was trying to make is about splitting up the code for reusability and discoverability, but it doesn't necessarily correspond to a module-level split.  The scenario you describe with multiple upstream sources is likely, and we'll also have multiple output files per module.

I'll still try to do encapsulation and information hiding but this is normal procedure and doesn't require special mention in the README.

Reply:
Perfect. Now that we have a similar view, I will approve the PR.